great_expectations/cli/project.py

```diff
@@ -66,13 +66,15 @@ def project_upgrade(ctx: click.Context) -> None:
         sys.exit(1)
 
 
-def do_config_check(target_directory: str) -> Tuple[bool, str, Optional[DataContext]]:
+def do_config_check(
+    target_directory: Union[str, None]
```

Comment:
This and other similar changes were made because the caller of the method had a value that could have been none.
If needed I also made changes to the downstream methods as needed.
Example: `send_usage_message` in usage stats.

Reply:
Why not `Optional[str]`?

Reply:
Because the word `optional` can be confusing to developers.

It isn't "optional"... it could be `None`, but it's never "optional".
`Optional` makes sense for keyword arguments that have a default.

https://fastapi.tiangolo.com/python-types/#using-union-or-optional

Reply:
But to clarify for anyone else.
`Union[str, None]` and `Optional[str]` are the same thing to the type-checker.
This is just a matter of convention/style.